src/lua/LuaPiGui.cpp

```diff
@@ -1156,49 +1168,102 @@ static InputBindings::KeyBinding get_joy_button()
 		}
 	}
 
-	return InputBindings::KeyBinding{};
+	return bindings;
 }
 
 // FIXME: implement an "input binding mode" which listens to events from Pi::input to build input chords
 static int l_pigui_get_keybinding(lua_State *l)
 {
 	PROFILE_SCOPED()
 
-	InputBindings::KeyChord binding;
-	int key = 0;
+	auto bindState = LuaPull<InputBindings::KeyChord>(l, 1);
+	KeyBindings binds{};
+	int bind_count = 0;
+
+	// Collect the first MAX_KEYS keys, joystick buttons, mouse buttons
 
-	// FIXME: support key chording instead of scanning the list of currently held keys!
-	// pick the first key that's currently held down
-	// should there be a priority?
+	// keys
 	for (int i = 0; i < 512; i++) {
 		if (ImGui::GetIO().KeysDown[i]) {
 			// io.KeysDown uses scancodes, but we need keycodes.
-			key = SDL_GetKeyFromScancode(static_cast<SDL_Scancode>(i));
-			break;
+			binds[bind_count++] = SDL_GetKeyFromScancode(static_cast<SDL_Scancode>(i));
+			if (bind_count == MAX_BIND_KEYS) break;
 		}
 	}
 
 	// Escape is used to clear an existing binding
-	if (key == SDLK_ESCAPE) {
+	if (bind_count == 1 && binds[0].keycode == SDLK_ESCAPE) {
 		LuaPush(l, true);
 		lua_pushnil(l);
 		return 2;
 	}
 
-	// Check joysticks if no keys are held down
-	if (Pi::input->IsJoystickEnabled() && (key == 0 || (key >= SDLK_LCTRL && key <= SDLK_RGUI))) {
-		binding.activator = get_joy_button();
-	} else if (key != 0) {
-		binding.activator = InputBindings::KeyBinding(key);
+	// joysticks
+	if (Pi::input->IsJoystickEnabled() && bind_count < MAX_BIND_KEYS) {
+		KeyBindings joy_binds = get_joy_buttons();
+		for (int i = 0; joy_binds[i].Enabled() && bind_count < MAX_BIND_KEYS; ++i) {
+			binds[bind_count++] = joy_binds[i];
+		}
 	}
 
-	if (!binding.Enabled()) {
+	// mouse
+	if (bind_count < MAX_BIND_KEYS) {
+		for (int i = 0; i < 5; i++) {
+			if (ImGui::GetIO().MouseDown[i]) {
+				// imgui mouse button numbering different from sdl
+				const uint8_t imgui_to_sdl[] = { 1, 3, 2, 4, 5 };
```

Comment:
Instead of an explicit array here, you should be able to directly use `i + 1` to get the SDL key binding index.

Reply:
@Web-eWorks note that the numbers in the array are not exactly in order. `i + 1` will confuse the middle and right mouse buttons.

Reply:
This is what I get for reviewing at 3am, sorry. Ignore this one!